src/dbally/views/pandas_base.py

```diff
@@ -63,11 +66,16 @@ async def build_filter_node(self, node: syntax.Node) -> pd.Series:
 
     def execute(self, dry_run: bool = False) -> ExecutionResult:
         """
-        Executes the view and returns the results. The results are filtered based on the applied filters.
+        Executes the view and returns the results. The results are filtered based on the applied filters.\
+        Be aware that before running this method you need to execute `apply_filters`
 
-        :param dry_run: If True, the method will only return the mask that would be applied to the dataframe
+        Args:
+            dry_run: If True, the method will only add `context` field to the `ExecutionResult` with the\
+            mask that would be applied to the dataframe
 
-        :return: ExecutionResult object with the results and context information
+        Returns:
+            ExecutionResult object with the results, being a result of DataFrame.to_dict(orient="record")\
```

Comment:
The fact that we've used `DataFrame.to_dict(orient="record")` is definitely an implementation detail. It's in the same format as `ExecutionResult` object from all other view classes. How we obtained it is not important.